src/Game/zLasso.cpp

```diff
@@ -1,15 +1,174 @@
+#include "iModel.h"
 #include "zLasso.h"
+#include "xEnt.h"
+#include "xSnd.h"
+#include "xstransvc.h"
+#include "xString.h"
+#include "xVec3.h"
 
 #include <types.h>
 
+// lbl_803CDAE0-_SDA2_BASE_(r2), -0x6ea0(r2)
+extern float32 zLasso_f_0_0;
+// lbl_803CDAE4-_SDA2_BASE_(r2), -0x6e9c(r2)
+extern float32 zLasso_f_1_0;
+// lbl_803CDAF0-_SDA2_BASE_(r2),
+extern float32 zLasso_f_30_0;
+// lbl_803CDAF4-_SDA2_BASE_(r2),
+extern float32 zLasso_2pi_6_28318548203;
+// lbl_803CDAF8-_SDA2_BASE_(r2),
+extern float32 zLasso_f_2_0;
+// lbl_803CDAFC-_SDA2_BASE_(r2),
+extern float32 zLasso_pi_3_14159274101;
+// lbl_803CDB00-_SDA2_BASE_(r2),
+extern float32 zLasso_f_3_0;
+// lbl_803CDB04-_SDA2_BASE_(r2),
+extern float32 zLasso_f_2_09439516068;
+// lbl_803CDB08-_SDA2_BASE_(r2),
+extern float32 zLasso_f_4_18879032135;
+// lbl_803CDB0C-_SDA2_BASE_(r2),
+extern float32 zLasso_f_0_0250000003725;
+// lbl_803CDB10-_SDA2_BASE_(r2),
+extern float32 zLasso_f_43_0;
+// lbl_803CDB14-_SDA2_BASE_(r2),
+extern float32 zLasso_f_0_2000000298;
+// lbl_803CDB18-_SDA2_BASE_(r2),
+extern float32 zLasso_f_0_75;           // 1/4
+// lbl_803CDB1C-_SDA2_BASE_(r2),
+extern float32 zLasso_f_0_666666686535; // 2/3
+// lbl_803CDB20-_SDA2_BASE_(r2),
+extern float32 zLasso_f_0_333333343267; // 1/3
+// lbl_803CDB24-_SDA2_BASE_(r2),
+extern float32 zLasso_f_0_418879032135;
+// lbl_803CDB28-_SDA2_BASE_(r2),
+extern float32 zLasso_f_15_0;
+// lbl_803CDB2C-_SDA2_BASE_(r2),
+extern float32 zLasso_f_9_99999974736en6; // 9.99999974738e-06
+
+#if 0
+typedef void* HANDLE;
+HANDLE gLastAsset = 0; // lbl_803CB8D8-_SDA_BASE_(r13), -0x7028(r13)
+// 00 00 00 01 00 00 00 00
+volatile int gNumStructs = 0; // lbl_803CB8DC-_SDA_BASE_(r13), -0x7024(r13)
+char *gRenderPtr; // lbl_803CB8E4-_SDA_BASE_(r13), -0x701c(r13) 
+extern char gRenderArr[];
+
+extern const char rope_labels[0x18]; // lbl_8025CD18
+
+// lbl_803CB8DC-_SDA_BASE_(r13), -0x7024(r13)
+extern const uint32 zLasso_i_4294967296; // 4294967296 (number of int32 values)
+
+typedef struct _x {
+    xEnt *pEnt;
+    int ctr;
+    xModelInstance *pModel;
+    xAnimState *pState;
+    void *ptrs[14];
+} STRUCT_A;
+extern void initVertMap(xModelInstance **a); // parameter should be zLassoGuide *
+extern void bakeMorphAnim(RpGeometry *a, void *b);
+
+typedef struct _y {
+    STRUCT_A Four[3];
+    int pad[2];
+} STRUCT_B;
+
+STRUCT_B gEntities[0xd00]; // lbl_802E5788, -0x7fd2
+#endif
```

Comment:
I think you should remove all of this junk that's excluded from the file already. It looks more like a madman's notes than helpful documentation or comments. Also, structs in this project are defined in header files and included, not within C++ files.

Reply:
We don't know what these structs are supposed to be. They are placeholders until we figure out the actual data.

Reply:
How have you surmised what you have now? There isn't any information about this struct in the DWARF data? `gEntities` isn't a global variable in the `symbols.txt`, which means it doesn't exist in the source code, so I'm not sure what prompted you to make it.